client/src/unifyfs-sysio.c

```diff
@@ -1606,15 +1613,16 @@ int UNIFYFS_WRAP(lio_listio)(int mode, struct aiocb* const aiocb_list[],
         rc = unifyfs_gfid_read_reqs(reqs, reqcnt);
         if (rc != UNIFYFS_SUCCESS) {
             /* error reading data */
-            ret = -1;
+            ret = rc;
```

Comment:
@sandrain , can ``unifyfs_gfid_read_reqs`` return something other than -1 on error?  Regardless, we want to be sure the ``lio_listio`` wrapper only returns -1.  To be clearer, I think it's best to leave this in as setting ``ret = -1``.

Reply:
Nevermind, now I see the other code below.